### PR TITLE
HW-Lottery

### PR DESCRIPTION
**Pull request by CodeWebWeaver**
Created requested classes and implemented method getRandomBall() in Lottery class

### CODE REVIEW

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 1]
```diff
@@ -1,7 +1,11 @@
...
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    private final Random randomColor = new Random();
+
+    public Colors getRandomColor() {
+        return Colors.values()[randomColor.nextInt(Colors.values().length)];
```
Comment (andrii-hoienko) [inline src/main/java/core/basesyntax/ColorSupplier.java:9]: Overloaded expression, please introduce some variables

File: src/main/java/core/basesyntax/ColorSupplier.java [hunk 2]
```diff
@@ -1,7 +1,11 @@
 package core.basesyntax;
 
+import java.util.Random;
+
 public class ColorSupplier {
-    public String getRandomColor() {
-        return null;
+    private final Random randomColor = new Random();
```
Comment (andrii-hoienko) [inline src/main/java/core/basesyntax/ColorSupplier.java:6]: variable randomColor should contain color from its name, I think you should rename it

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Added some comments, hope you can manage them fast)

File: src/main/java/core/basesyntax/Colors.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package core.basesyntax;
+
+public enum Colors {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Colors.java:3]:
```suggestion
public enum Color {
```

File: src/main/java/core/basesyntax/Lottery.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_BALL_NUMBER = 100;
+    private final Random randomNumberGenerator = new Random();
+
+    public Ball getRandomBall() {
+        int randomNumber = randomNumberGenerator.nextInt(MAX_BALL_NUMBER);
+        return new Ball(randomNumber);
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Lottery.java:11]: what about color?

File: src/main/java/core/basesyntax/Ball.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Ball {
+    public final Colors color;
+    public final int number;
+    private final ColorSupplier colorSupply = new ColorSupplier();
+
+    Ball(int number) {
+        this.color = colorSupply.getRandomColor();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Ball.java:9]: move this logic into getRandomBall() method

and add access modifier to constructor

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
code style check failed
run `mvn clean package` before push

File: src/main/java/core/basesyntax/Ball.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Ball {
+    private final Color color;
+    private final int number;
+
+    Ball(Color color, int number) {
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Ball.java:7]: add access modifier

File: src/main/java/core/basesyntax/Lottery.java [hunk 7]
```diff
@@ -0,0 +1,15 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_BALL_NUMBER = 100;
+    private final Random randomNumberGenerator = new Random();
+    private final ColorSupplier colorSupply = new ColorSupplier();
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Lottery.java:8]:
```suggestion
    private final ColorSupplier colorSupplier = new ColorSupplier();
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -1,18 +1,18 @@
...
-      - uses: actions/checkout@v2
-      - name: Set up JDK 17
-        uses: actions/setup-java@v2
-        with:
-          java-version: '17'
-          distribution: 'adopt'
-          cache: maven
-      - name: Build with Maven
-        run: mvn --batch-mode --update-snapshots verify
+name: Java CI
```
Comment (andrii-hoienko) [inline .github/workflows/ci.yml:1]: Please dont push ( and dont change if it is not necessary) files which is not related to homework

File: README.md [hunk 9]
```diff
@@ -1,18 +1,18 @@
...
-#### [Try to avoid these common mistakes, while solving task](https://mate-academy.github.io/jv-program-common-mistakes/java-core/oop/lottery)
+# jv-lottery
+
+Let's create a simple lottery that will return balls with a random color and random number.
+
+Your tasks are:
+- create `Ball` class with the following fields: `color` and `number` and override `toString()` method
+- create enum with different colors
+- implement method `getRandomColor()` in class `ColorSupplier`. 
+This method should return a random color, based on the enum values (use `Random` class for this)
```
Comment (andrii-hoienko) [inline README.md:9]: Please dont push ( and dont change if it is not necessary) files which is not related to homework

File: checklist.md [hunk 10]
```diff
@@ -1,206 +1,206 @@
-## Common mistakes (jv-lottery)
-
-#### Please don't add redundant empty lines to your code.
-We don't need them after class declaration or method signature.
-* Bad example:
-```
-public class Main {
```
Comment (andrii-hoienko) [inline checklist.md:7]: Please dont push ( and dont change if it is not necessary) files which is not related to homework

File: pom.xml [hunk 11]
```diff
@@ -1,68 +1,68 @@
...
-            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
-        </maven.checkstyle.plugin.configLocation>
-    </properties>
-
-    <dependencies>
-        <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.12</version>
-            <scope>test</scope>
```
Comment (andrii-hoienko) [inline pom.xml:26]: Please dont push ( and dont change if it is not necessary) files which is not related to homework

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Everything is good reagardind solution, but there are some unnecessary files in commit

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request closed without merging**